src/common/mfu_daos.c

```diff
@@ -1313,3 +1330,2195 @@ int mfu_flist_copy_daos(
+    if (sizes == NULL) {
+        rc = ENOMEM;
+        goto out;
+    }
+    val_buf = malloc(num_attrs * sizeof(void*));
+    if (val_buf == NULL) {
+        rc = ENOMEM;
+        goto out;
+    }
+
+    rc = daos_cont_get_attr(cont, num_attrs,
+                (const char* const*)names,
+                NULL, sizes, NULL);
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to get attr size");
+        rc = 1;
+        goto out;
+    }
+    for (j = 0; j < num_attrs; j++) {
+        val_buf[j] = malloc(sizes[j]);
+        if (val_buf[j] == NULL) {
+            rc = ENOMEM;
+            goto out;
+        }
+    }
+    rc = daos_cont_get_attr(cont, num_attrs,
+                            (const char* const*)names,
+                            (void * const*)val_buf, sizes,
+                            NULL);
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "failed to get attr value");
+        rc = 1;
+        goto out;
+    }
+    /* write user attrs to hdf5 file */
+    for (i = 0; i < num_attrs; i++) {
+        /* TODO: remove debug code 
+        printf("attr size: %d\n", (int)sizes[i]);
+        printf("attr name: %s\n", (char*)names[i]);
+        char str[256];
+        snprintf(str, sizes[i] + 1, "%s", (char *)val_buf[i]);
+        printf("attr val: %s\n", str);
+        */
+        hdf5->attr_dims[0] = 1;
+        hdf5->attr_dtype = H5Tcreate(H5T_OPAQUE, sizes[i]);
+        if (hdf5->attr_dtype < 0) {
+            MFU_LOG(MFU_LOG_ERR, "failed to create user attr type");
+            rc = 1;
+            goto out;
+        }
+        hdf5->attr_dspace = H5Screate_simple(1, hdf5->attr_dims,
+                                             NULL);
+        if (hdf5->attr_dspace < 0) {
+            MFU_LOG(MFU_LOG_ERR, "failed to create version attribute");
+            rc = 1;
+            goto out;
+        }
+        hdf5->usr_attr = H5Acreate2(hdf5->file,
+                                    names[i],
+                                    hdf5->attr_dtype,
+                                    hdf5->attr_dspace,
+                                    H5P_DEFAULT,
+                                    H5P_DEFAULT);
+        if (hdf5->usr_attr < 0) {
+            MFU_LOG(MFU_LOG_ERR, "failed to create user attribute name");
+            rc = 1;
+            goto out;
+        }   
+        status = H5Awrite(hdf5->usr_attr, hdf5->attr_dtype,
+                          val_buf[i]);
+        if (status < 0) {
+            MFU_LOG(MFU_LOG_ERR, "failed to write attribute");
+            rc = 1;
+            goto out;
+        }
+        status = H5Aclose(hdf5->usr_attr);
+        if (status < 0) {
+            MFU_LOG(MFU_LOG_ERR, "failed to close user attribute");
+            rc = 1;
+            goto out;
+        }
+    }
+out:
+    if (buf != NULL)
+        mfu_free(&buf);
+    for (j = 0; j < num_attrs; j++) {
+        mfu_free(&val_buf[j]);
+    }
+    if (val_buf != NULL) {
+        mfu_free(&val_buf);
+    }
```

Comment:
It's valid to call ``mfu_free`` even when ``val_buf == NULL``, so you can drop the if check on these to simplify like so:
```suggestion
    mfu_free(&val_buf);
```